geoarrow-pyarrow/src/geoarrow/pyarrow/io.py

```diff
@@ -36,7 +40,8 @@ def read_pyogrio_table(*args, **kwargs):
 
     meta, table = read_arrow(*args, **kwargs)
 
-    # Maybe not always true? meta["geometry_name"] is occasionally `""`
+    # When meta["geometry_name"] is `""`, the geometry column name is wkb_geometry
+    # in GDAL's Arrow output
```

Comment:
I agree renaming `wkb_geometry` to `geometry` makes sense. Also, instead of inferring the name, you should be able to check which field has extension type `ogc.wkb`, which I think is probably more reliable than guessing the name `wkb_geometry`